guacml/pipeline.py

```diff
@@ -0,0 +1,33 @@
+from guacml.dataset import Dataset
+from guacml.preprocessing.column_analyzer import ColumnAnalyzer
+
+
+class Pipeline:
+
+    def __init__(self, guac, model_name):
+        mr = guac.model_results[model_name]
+        self.name = model_name
+        self.tree = guac.tree.get_subtree_upto(model_name)
+        self.config = self.tree.config
+        self.config['column_types'] = mr.metadata.type.to_dict()
+        self.logger = guac.logger
+        self.model = mr.model.copy(self.config)
+        self.features = list(mr.features)
+
+    def transform(self, test_set):
+        step_name = self.tree.root_name
+        metadata = ColumnAnalyzer(self.config, self.logger).analyze(test_set)
```

Comment:
I think we should serialize the metadata of the `ColumnAnalyzer` that was used to train the model for the following reasons:

- it seems like an unnecessary overhead to reanalyse the dataset at prediction time (even if its just a small one). For production use cases we want the prediction step to be as fast as possible
- if the dataset that is passed into the prediction step is small (imagine one single row) the results of `analyse` might be vastly different form the `analyse` result that we had during training of the model.
- if the given dataset would rightfully lead to a different set of metadata we should raise an error. Lets say there was a string supplied where we expected an integer. In cases like this we should tell the user that we expected something else in this column. There are cases where it is not so easy though. One that comes to mind is: What if we see a new categorical value on prediction time? Should we ignore it or report an error?

WDYT?